lib/rubocop/cop/rails/has_many_or_has_one_dependent.rb

```diff
@@ -60,7 +60,7 @@ class HasManyOrHasOneDependent < Base
           (block
             (send nil? :with_options
               (hash $...))
-            (args) ...)
+            (args ...) ...)
```

Comment:
Would the condition of 0 or 1 suffice?

```suggestion
            (args _?) ...)
```

Reply:
In the examples I have tested for, yes, it would suffice.

However, I believe it would be perfectly valid Ruby to define an arbitrary number of parameters for the block:
```rb
with_options dependent: :destroy do |model, foo, bar|
  model.has_many :things
end
```
It feels to me that we would not want this cop to flag an offence in this scenario. WDYT?

Reply:
If it occurs in practice, please add it to the test cases. Otherwise, please update the product code.

Reply:
Thanks. This is highly unlikely to ever occur in practice. I will update the pattern.